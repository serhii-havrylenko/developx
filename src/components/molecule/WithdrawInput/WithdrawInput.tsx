import TextField from '@material-ui/core/TextField';
import * as React from 'react';
import styled from 'styled-components';

import Button from '#atom/Button';

const WithdrawInputWrapper = styled.div`
  display: flex;

  & > * {
    align-self: normal;
  }

  & > *:not(:first-child) {
    margin-left: 10px;
  }
`;

export interface WithdrawInputProps {
  amountRef: React.RefObject<HTMLInputElement>;
  onWithdrawClick: () => void;
}

const WithdrawInput: React.SFC<WithdrawInputProps> = ({
  onWithdrawClick,
  amountRef,
}) => (
  <WithdrawInputWrapper>
    <TextField label="Amount" type="number" inputRef={amountRef} />
    <Button onClick={onWithdrawClick}>Withdraw</Button>
  </WithdrawInputWrapper>
);

export default WithdrawInput;
