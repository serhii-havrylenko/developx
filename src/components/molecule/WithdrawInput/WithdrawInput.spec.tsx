import { mount } from 'enzyme';
import * as React from 'react';

import WithdrawInput from './WithdrawInput';

jest.mock('@material-ui/core/Button');
jest.mock('@material-ui/core/TextField');

describe('<WithdrawInput /> tests', () => {
  const props = {
    onWithdrawClick: jest.fn(),
    amountRef: React.createRef() as React.RefObject<HTMLInputElement>,
  };

  test('should match snapshot', () => {
    const tree = mount(<WithdrawInput {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
