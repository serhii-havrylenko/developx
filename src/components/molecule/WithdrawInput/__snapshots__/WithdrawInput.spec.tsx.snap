// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithdrawInput /> tests should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > * {
  -webkit-align-self: normal;
  -ms-flex-item-align: normal;
  align-self: normal;
}

.c0 > *:not(:first-child) {
  margin-left: 10px;
}

<WithdrawInput
  amountRef={
    Object {
      "current": null,
    }
  }
  onWithdrawClick={[MockFunction]}
>
  <styled.div>
    <div
      className="c0"
    >
      <TextField
        inputRef={
          Object {
            "current": null,
          }
        }
        label="Amount"
        required={false}
        select={false}
        type="number"
        variant="standard"
      />
      <Button
        onClick={[MockFunction]}
      >
        <WithStyles(Button)
          color="primary"
          onClick={[MockFunction]}
        />
      </Button>
    </div>
  </styled.div>
</WithdrawInput>
`;

exports[`<WithdrawInput /> tests should match snapshot 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > * {
  -webkit-align-self: normal;
  -ms-flex-item-align: normal;
  align-self: normal;
}

.c0 > *:not(:first-child) {
  margin-left: 10px;
}

<WithdrawInput
  amountRef={
    Object {
      "current": null,
    }
  }
  onWithdrawClick={[MockFunction]}
>
  <styled.div>
    <div
      className="c0"
    >
      <TextField
        inputRef={
          Object {
            "current": null,
          }
        }
        label="Amount"
        required={false}
        select={false}
        type="number"
        variant="standard"
      />
      <Button
        onClick={[MockFunction]}
      >
        <WithStyles(Button)
          color="primary"
          onClick={[MockFunction]}
        />
      </Button>
    </div>
  </styled.div>
</WithdrawInput>
`;
